src/Driver/PDO/MySQL/Driver.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Facile\DoctrineMySQLComeBack\Doctrine\DBAL\Driver\PDO\MySQL;
+
+use Doctrine\DBAL\Driver\AbstractMySQLDriver;
+use Doctrine\DBAL\Driver\PDO;
+use Facile\DoctrineMySQLComeBack\Doctrine\DBAL\Driver\ServerGoneAwayExceptionsAwareInterface;
+use Facile\DoctrineMySQLComeBack\Doctrine\DBAL\Driver\ServerGoneAwayExceptionsAwareTrait;
+
+/**
+ * Class Driver.
+ */
+class Driver extends AbstractMySQLDriver implements ServerGoneAwayExceptionsAwareInterface
+{
+    use ServerGoneAwayExceptionsAwareTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function connect(array $params, $username = null, $password = null, array $driverOptions = [])
+    {
+        return new PDO\Connection(
+            $this->constructPdoDsn($params),
+            $username ?? $params['user'] ?? '',
+            $password ?? $params['password'] ?? '',
+            $driverOptions
+        );
+    }
+
+    /**
+     * Constructs the MySql PDO DSN.
+     *
+     * @param mixed[] $params
+     *
+     * @return string The DSN.
+     */
+    protected function constructPdoDsn(array $params): string
```

Comment:
Why not private?

Reply:
I didn't want to change the visibility of the parent method.

Reply:
I don't see a parent method... Is it there on a specific dependency version?

Reply:
You're right, I've just copied and pasted from the original driver class, but we can make the class final and change it to private.